segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/Ledgers.java

```diff
@@ -82,14 +93,20 @@ static LedgerHandle openFence(long ledgerId, BookKeeper bookKeeper, BookKeeperCo
      * @param ledgerId   The Id of the Ledger to open.
      * @param bookKeeper A references to the BookKeeper client to use.
      * @param config     Configuration to use.
-     * @return A LedgerHandle for the newly opened ledger.
+     * @return A ReadHandle for the newly opened ledger.
      * @throws DurableDataLogException If an exception occurred. The causing exception is wrapped inside it.
      */
-    static LedgerHandle openRead(long ledgerId, BookKeeper bookKeeper, BookKeeperConfig config) throws DurableDataLogException {
+    static ReadHandle openRead(long ledgerId, BookKeeper bookKeeper, BookKeeperConfig config) throws DurableDataLogException {
         try {
             return Exceptions.handleInterruptedCall(
-                    () -> bookKeeper.openLedgerNoRecovery(ledgerId, LEDGER_DIGEST_TYPE, config.getBKPassword()));
-        } catch (BKException bkEx) {
+                    () -> FutureUtils.result(bookKeeper
+                            .newOpenLedgerOp()
+                    .withLedgerId(ledgerId)
+                    .withDigestType(LEDGER_DIGEST_TYPE)
+                    .withPassword(config.getBKPassword())
+                    .withRecovery(false)
+                    .execute()));
+        } catch (Exception bkEx) {
```

Comment:
Same question about the exception handler.

Reply:
Updated with BK_EXCEPTION_HANDLER